doc/source/cluster/kubernetes/examples/rayjob-kueue-gang-scheduling.md

```diff
@@ -0,0 +1,249 @@
+    --num-nodes=1 --min-nodes 0 --max-nodes 1 --enable-autoscaling \
+    --zone=us-west1-b --machine-type e2-standard-4 --cluster-version 1.29
+```
+
+Create a GPU node pool with the "enable-queued-provisioning" option enabled:
+```bash
+gcloud beta container node-pools create gpu-node-pool \
+  --accelerator type=nvidia-l4,count=1,gpu-driver-version=latest \
+  --enable-queued-provisioning \
+  --reservation-affinity=none  \
+  --zone us-west1-b \
+  --cluster kuberay-gpu-cluster \
+  --num-nodes 0 \
+  --min-nodes 0 \
+  --max-nodes 10 \
+  --enable-autoscaling \
+  --machine-type g2-standard-4
+```
+
+:::{note}
+"enable-queued-provisioning" is only available on versions 1.28+
+:::
+
+
+## Install the KubeRay Operator
+
+Follow [Deploy a KubeRay operator](kuberay-operator-deploy) to install the latest stable KubeRay operator from the Helm repository.
+The KubeRay operator Pod must be on the CPU node if you set up the taint for the GPU node pool correctly.
+
+## Install Kueue
+
+Install Kueue with the ProvisioningRequest API enabled.
+```
+kubectl apply --server-side -f https://github.com/kubernetes-sigs/kueue/releases/download/v0.6.0-rc.2/manifests-alpha-enabled.yaml
+```
+
+See [Kueue Installation](https://kueue.sigs.k8s.io/docs/installation/#install-a-released-version) for more details on installing Kueue.
+
+## Configure Kueue for gang scheduling
+
+Next, configure Kueue for gang scheduling. Kueue leverages the ProvisioningRequest API for two key tasks:
+1. Dynamically adding new nodes to the cluster when more resources are needed for a job.
+2. Blocking the admission of new jobs while waiting for sufficient resources to become available.
+
+See [How ProvisioningRequest works](https://cloud.google.com/kubernetes-engine/docs/how-to/provisioningrequest#how-provisioningrequest-works) for more details.
+
+### Create Kueue Resources
+
+This manifest creates the following resources:
+* [ClusterQueue](https://kueue.sigs.k8s.io/docs/concepts/cluster_queue/): defines quotas and fair sharing rules
+* [LocalQueue](https://kueue.sigs.k8s.io/docs/concepts/local_queue/): a namespaced queue (belonging to a tenant) that references a ClusterQueue
+* [ResourceFlavor](https://kueue.sigs.k8s.io/docs/concepts/resource_flavor/): defines what resources are available in the cluster (typically from Nodes)
+* [AdmissionCheck](https://kueue.sigs.k8s.io/docs/concepts/admission_check/): a mechanism allowing components to influence timing of workload admission
+
+```yaml
+# kueue-resources.yaml
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: ResourceFlavor
+metadata:
+  name: "default-flavor"
+---
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: AdmissionCheck
+metadata:
+  name: rayjob-gpu
+spec:
+  controllerName: kueue.x-k8s.io/provisioning-request
+  parameters:
+    apiGroup: kueue.x-k8s.io
+    kind: ProvisioningRequestConfig
+    name: rayjob-gpu-config
+---
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: ProvisioningRequestConfig
```

Comment:
@kevin85421 yes that's correct, Kueue is always in gang mode in that it will only admit workloads when all of it's resources are available.

However, without autoscaling and provisioning request, it is up to the cluster admin to properly configure quotas in ClusterQueue to reflect the capacity available in the cluster. With autoscaling and provisioning request, gang scheduling is automatically handled and the quota can be configured to "infinity". Both approaches work and have trade-offs, for this guide I decided to use the autoscaling approach so we don't have to configure static quotas